velox/exec/tests/MultiFragmentTest.cpp

```diff
@@ -378,6 +379,22 @@ TEST_F(MultiFragmentTest, partitionedOutput) {
 
     ASSERT_TRUE(waitForTaskCompletion(leafTask.get())) << leafTask->taskId();
   }
+
+  // Test asynchronously deleting task buffer (due to abort from downstream)
+  {
+    auto leafTaskId = makeTaskId("leaf", 0);
+    auto leafPlan = PlanBuilder()
+                        .values(vectors_)
+                        .partitionedOutput({}, 1, {"c0", "c1"})
+                        .planNode();
+    auto leafTask = makeTask(leafTaskId, leafPlan, 0);
+    Task::start(leafTask, 4);
+    auto bufferMgr = PartitionedOutputBufferManager::getInstance().lock();
+    // Deletes the results asynchronously to simulate abort from downstream
+    bufferMgr->deleteResults(leafTaskId, 0);
+
+    ASSERT_TRUE(waitForTaskCompletion(leafTask.get())) << leafTask->taskId();
```

Comment:
Just to clarify, this tests would fail without the fix, right?

Reply:
Yep.